### PR TITLE
Docs: autosnapshots notes around AWS EKS & S3 configuration requirments

### PR DESCRIPTION
**Pull request by aphorise**
Added requirement [notes to API](https://developer.hashicorp.com/vault/api-docs/system/storage/raftautosnapshots) & [Docs autosnapshot](https://developer.hashicorp.com/vault/docs/enterprise/automated-integrated-storage-snapshots) in the context AWS & EKS with S3 vs current text that just states limitations without note of what's needed instead.

Screenshot 2024-07-01 at 13 41 20

### CODE REVIEW

File: website/content/docs/enterprise/automated-integrated-storage-snapshots.mdx [hunk 1]
```diff
@@ -37,9 +37,10 @@ object store.
 
 <Note>
 
-Currently, Vault does not allow the use of AWS IAM Roles for EKS Service
-Accounts to authenticate to Amazon S3 buckets for the Automated Integrated
-Storage Snapshots.
+Vault does not allow the use of AWS IAM Roles for EKS Service Accounts to 
+authenticate to Amazon S3 buckets for the Automated Integrated Storage Snapshots. 
+The parameters: [`aws_access_key_id` & `aws_secret_access_key`](/vault/api-docs/system/storage/raftautosnapshots#aws_access_key_id) 
+must be set in the context of AWS EKS & S3 configuration.
```
Comment (schavis) [inline website/content/docs/enterprise/automated-integrated-storage-snapshots.mdx:43]:
```suggestion
Vault cannot use AWS IAM roles with EKS service accounts for authentication to
save automated integrated storage snapshots to Amazon S3 buckets. You must set
the [`aws_access_key_id`](/vault/api-docs/system/storage/raftautosnapshots#aws_access_key_id)
and  [`aws_secret_access_key`](/vault/api-docs/system/storage/raftautosnapshots#aws_secret_access_key)
parameters in the context of AWS EKS & S3 configuration.
```
Style correction: write in the active voice

File: website/content/api-docs/system/storage/raftautosnapshots.mdx [hunk 2]
```diff
@@ -67,6 +67,11 @@ environment variables or files on disk in predefined locations.
 
 #### storage_type=aws-s3
 
+~> ** Note:** Vault does not allow the use of AWS IAM Roles for EKS Service Accounts 
+   to authenticate to Amazon S3 buckets for the Automated Integrated Storage Snapshots. 
+   The parameters: [`aws_access_key_id` & `aws_secret_access_key`](/vault/api-docs/system/storage/raftautosnapshots#aws_access_key_id) 
+   must be set in the context of AWS EKS & S3 configuration.
```
Comment (schavis) [inline website/content/api-docs/system/storage/raftautosnapshots.mdx:73]:
```suggestion
<Note>

Vault cannot use AWS IAM roles with EKS service accounts for authentication to
save automated integrated storage snapshots to Amazon S3 buckets. You must set
the [`aws_access_key_id`](/vault/api-docs/system/storage/raftautosnapshots#aws_access_key_id)
and  [`aws_secret_access_key`](/vault/api-docs/system/storage/raftautosnapshots#aws_secret_access_key)
parameters in the context of AWS EKS & S3 configuration.

</Note>
```
Edited to match the other suggestion

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/enterprise/automated-integrated-storage-snapshots.mdx (hunk 1 above)
Comment (aphorise) [reply website/content/docs/enterprise/automated-integrated-storage-snapshots.mdx:43]: Thanks @schavis - I was trying to make very minimal changes to the original text but this correction does indeed seem more correct.

**Comment by aphorise** [comment]
I do not have merge capabilities - kindly merge this PR.

**Review by schavis — APPROVED** [review]

**Pull request merged**